### PR TITLE
Add go plugins feature

### PR DESCRIPTION
**Pull request by xianlezheng**
**What this PR does**:Ftr:Support go plugins with no changes to the project code

**Which issue(s) this PR fixes**:9

Fixes #

**Special notes for your reviewer**: 

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/filter/plugins/plugins.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package plugins
+
+import (
+	"errors"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/filter"
```
Comment (xiaoliu10) [inline pkg/filter/plugins/plugins.go:22]: split import

File: pkg/filter/plugins/plugins.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+type PluginsWithFunc struct {
+	Name     string
+	Priority int
+	fn       context.FilterFunc
+}
+
+// Prase api_config.yaml(pluginsGroup) to map[string][]PluginsWithFunc
+func InitPluginsGroup(groups []config.PluginsGroup, filePath string) {
+
+	if "" == filePath {
```
Comment (xiaoliu10) [inline pkg/filter/plugins/plugins.go:50]: if len(filePath) == 0 ?

File: sample/plugins/plugin_test.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package main
+
+import (
+	"github.com/dubbogo/dubbo-go-proxy/pkg/context"
```
Comment (xiaoliu10) [inline sample/plugins/plugin_test.go:20]: split import, thx

**Review by xiaoliu10 — APPROVED** [review]

File: pkg/filter/plugins/plugins.go [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+// InitPluginsGroup prase api_config.yaml(pluginsGroup) to map[string][]PluginsWithFunc
+func InitPluginsGroup(groups []config.PluginsGroup, filePath string) {
+
+	if "" == filePath || len(groups) == 0 {
+		return
+	}
+
+	// load file.so
+	pls, err := plugin.Open(filePath)
```
Comment (mark4z) [inline pkg/filter/plugins/plugins.go:55]: configs/plugins/plugin.so should not be included in the project beacuse it's a bin file. Maybe build it in workflows is a better way.

**Review by mark4z — CHANGES_REQUESTED** [review]
configs/plugins/plugin.so should not be included in the project beacuse it's a bin file. Maybe build it in workflows is a better way.

**Review by mark4z — APPROVED** [review]

**Review by mark4z — APPROVED** [review]

**Review by mark4z — APPROVED** [review]
LGTM & I will do it later

**Review by cityiron — APPROVED** [review]

**Pull request merged**